packages/apps/transfer/src/pages/transfer/cross-chain-transfer-finisher/index.tsx

```diff
@@ -1,128 +1,308 @@
+import { ChainwebNetworkId, IPollResponse } from '@kadena/chainweb-node-client';
+import { ContCommand } from '@kadena/client';
 import { Button, TextField } from '@kadena/react-components';
+import { ChainId } from '@kadena/types';
 
 import MainLayout from '@/components/Common/Layout/MainLayout';
-import { kadenaConstants } from '@/constants/kadena';
+import { chainNetwork } from '@/constants/network';
 import { useAppContext } from '@/context/app-context';
 import {
   StyledAccountForm,
   StyledCheckbox,
   StyledCheckboxLabel,
+  StyledErrorMessage,
   StyledFieldCheckbox,
+  StyledFinisherContent,
   StyledForm,
   StyledFormButton,
+  StyledInfoBox,
+  StyledInfoItem,
+  StyledInfoItemLine,
+  StyledInfoItemTitle,
+  StyledInfoTitle,
+  StyledResultContainer,
+  StyledShowMore,
   StyledToggleContainer,
+  StyledTotalChunk,
+  StyledTotalContainer,
 } from '@/pages/transfer/cross-chain-transfer-finisher/styles';
+import {
+  finishXChainTransfer,
+  ITransferResult,
+} from '@/services/cross-chain-transfer-finish/finish-xchain-transfer';
+import {
+  getTransferData,
+  ITransferDataResult,
+} from '@/services/cross-chain-transfer-finish/get-transfer-data';
+import { generateApiHost } from '@/services/utils/utils';
 import useTranslation from 'next-translate/useTranslation';
-import React, { FC, FormEventHandler, useState } from 'react';
+import React, { FC, useEffect, useState } from 'react';
+
+interface IPactResultError {
+  status: 'failure';
+  error: {
+    message: string;
+  };
+}
 
 const CrossChainTransferFinisher: FC = () => {
   const { t } = useTranslation('common');
   const { network } = useAppContext();
 
-  const chainNetwork: {
-    Mainnet: { server: string; network: string };
-    Testnet: { server: string; network: string };
-  } = {
-    Mainnet: {
-      server: kadenaConstants.MAINNET.API,
-      network: kadenaConstants.MAINNET.NETWORKS.MAINNET01,
-    },
-    Testnet: {
-      server: kadenaConstants.TESTNET.API,
-      network: kadenaConstants.TESTNET.NETWORKS.TESTNET04,
-    },
-  };
-
   const [requestKey, setRequestKey] = useState<string>('');
   const [kadenaXChainGas, setKadenaXChainGas] =
     useState<string>('kadena-xchain-gas');
```

Comment:
Setting the gas station as the sender should be a conscious choice, not the default behaviour. This is to have a minor barrier in using the gas station. Also, when the gas station is empty (which is possible) there should be a way to have a custom sender